tests/test_cases/test_discovery/test_discovery.py

```diff
@@ -461,7 +461,7 @@ def type_check_verilog(dut):
     test_handles = [
         (dut.stream_in_ready, "GPI_REGISTER"),
         (dut.register_array, "GPI_ARRAY"),
-        (dut.NUM_OF_MODULES, "GPI_PARAMETER"),
+        (dut.NUM_OF_MODULES, "GPI_INTEGER"),
```

Comment:
Are there any tests for the other cases such as GPI_REAL, GPI_STRING?
Which simulators have been tested with this change?

Reply:
Icarus, GHDL, and Active-HDL. I suppose I can test Questa real quick. And I don't think there are other tests, they are not exhaustive, I'll add some.

Reply:
Added constant integers, strings, and reals to this test. It should be tested using Icarus, Verilator, and Active-HDL.